### PR TITLE
Fix check for dns nameserver grain

### PR DESCRIPTION
**Pull request by drbachler**
Ok, so this check for `grains['dns']['nameserver']` exploded in my face with `Jinja variable 'dict object' has no attribute 'nameservers' saltstack` while trying to use this formula in the new PT salt environment.

This was due to in the old environment there is a grain `grains['dns']['nameserver']` and in the new environment the grain is called `grains['dns']['name_server']`

You can't check if a key in a dict is set that does not exist. So lets check if that key exists instead. Then set a variable to that key's contents and use that instead.

So this fixes that issue.
This was tested to make sure that we produce the same output as to not cause a restart of haproxy service where we already use this.

TL;DR:
This check for `grains['dns']['nameserver']` did not work as intended, this fixes that.

### CODE REVIEW

File: haproxy/files/haproxy.cfg [hunk 1]
```diff
@@ -33,10 +33,15 @@ global
   ssl-default-server-ciphers ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS
   ssl-default-server-options no-sslv3 no-tls-tickets
 
-  {% if grains['dns']['nameservers'] %}
+  {% if 'nameservers' in grains['dns'] -%}
```
Comment (pcn) [inline haproxy/files/haproxy.cfg:36]: There's a different idiom you can use if you know you have 2 possibilities:

`{% set nameservers = salt['grains.get']('dns:nameservers', salt['grains.get']('dns:name_servers', ["NAME_SERVER_GRAIN_NOT_FOUND"])) %}`

Then you can check `{% if nameservers[0] != 'NAME_SERVER_GRAIN_NOT_FOUND' %}`

Comment (drbachler) [reply haproxy/files/haproxy.cfg]: Yeah, 
I implemented this pattern and it works as expected.
Thanks for pointing that out, I wasn't really happy with how my first attempt looked.
Where did you find out about this? Any documentation? I sometimes find it hard to find documentation for what I want to do with Jinja, especially in relation to salt stuff.

**Review by pcn — APPROVED** [review]
LGTM

**Review by librato-peter — APPROVED** [review]
:+1: As explained in comments.
